src/App.tsx

```diff
@@ -1,156 +1,134 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useContext, useEffect, useRef, useState } from 'react';
+import { TodoMethod, TodosContext } from './component/TodosContext';
+import cn from 'classnames';
+import { SortType } from './component/types/types';
+import { Todo } from './component/todo';
 
 export const App: React.FC = () => {
+  const [title, setTitle] = useState('');
+
+  const methods = useContext(TodoMethod);
+  const { todos, sorted, setSorted } = useContext(TodosContext);
+
+  const field = useRef<HTMLInputElement>(null);
+
+  const activeTodos = todos.filter(todo => !todo.completed);
+  const visibleTodos = [...todos].filter(todo => {
+    switch (sorted) {
+      case SortType.Active:
+        return !todo.completed;
+      case SortType.Completed:
+        return todo.completed;
+      default:
+        return -1;
```

Comment:
move filter inside switch case not vice versa to avoid extra iteration in default case